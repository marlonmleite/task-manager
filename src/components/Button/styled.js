import styled from 'styled-components'

export const ButtonStyled = styled.button`
  padding: 0.4375rem 0.75rem;
  line-height: 1.25rem;
  font-size: 0.875rem;
  background-color: ${({ theme, color }) => theme.colors[color]};
  color: ${({ theme, color }) => theme.button.text[color]};
  display: inline-block;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  user-select: none;
  border: 1px solid ${({ theme, color }) => theme.colors[color]};
  border-radius: 2px;
  position: relative;

  :focus,
  :hover {
    text-decoration: none;
    background-image: linear-gradient(transparent, rgba(0,0,0,.05) 40%, rgba(0,0,0,.1));
  }

  :focus {
    outline: none;
  }

  .disabled,
  :disabled {
    opacity: .65;
    box-shadow: none;
  }

  :not([disabled]):not(.disabled) {
    cursor: pointer;
  }

  ${({ loading }) => loading && `
    > span {
      visibility: hidden;
    }
  `};
`

export const LoadingContainer = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`
