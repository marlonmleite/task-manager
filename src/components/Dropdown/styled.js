import styled, { css } from 'styled-components'

export const Dropdown = styled.div`
  position: relative;

  > button {
    width: 100%;
  }
`

export const DropdownContent = styled.div`
  position: absolute;
  min-width: 200px;
  color: #89898a;
  background-color: #fff;
  border-radius: 2px;
  margin-top: 2px;
  padding: 3px 5px;
  top: 100%;
  left: 0;
  z-index: 99;
  display: flex;
  flex-direction: column;
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.14), 0 3px 14px 2px rgba(0,0,0,0.12), 0 5px 5px -3px rgba(0,0,0,0.2);

  ${({ right }) => right && css`
    left: auto;
    right: 0;
  `};
`

export const DropdownButton = styled.div`
  display: flex;
  user-select: none;
  align-items: center;
  justify-content: space-between;

  > span {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
`

export const DropdownOption = styled.div`
  width: 100%;
  position: relative;
  padding: 0 1.25rem 0 1.5rem;
  box-sizing: border-box;
  color: #222;
  height: 36px;
  line-height: 36px;
  font-size: 14px;

  :hover {
    background-color: #eee;
    cursor: pointer;
  }

  ${({ selected }) => selected && css`
    :before {
      content: '';
      border-left: 2px solid #666;
      border-bottom: 2px solid #666;
      width: 10px;
      height: 6px;
      margin-top: -7px;
      position: absolute;
      left: 5px;
      top: 50%;
      transform: rotate(-45deg);
    }
  `};
`

export const CaretStyled = styled.div`
  margin-left: 10px;
  width: 0px;
  height: 0px;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-top: 5px solid ${({ color, theme }) => theme.button.text[color]};
`
