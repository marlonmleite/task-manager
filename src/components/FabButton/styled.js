import styled from 'styled-components'
import Button from 'components/Button'

const FabButton = styled(Button)`
  padding: 1rem;
  border-radius: 100%;
  width: 56px;
  height: 56px;
  position: fixed;
  right: 20px;
  bottom: 25px;
  box-shadow: 0 0 4px rgba(0,0,0,.04), 0 4px 8px rgba(0,0,0,.08);
`

export default FabButton
