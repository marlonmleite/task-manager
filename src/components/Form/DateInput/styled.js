import styled from 'styled-components'

const Date = styled.div`
  .react-datepicker {
    border-radius: 2px;
    font-family: inherit;
  }
`

export default Date
