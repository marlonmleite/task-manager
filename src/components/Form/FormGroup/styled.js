import styled, { css } from 'styled-components'

const form = {
  labelOffset: '25px',
  paddingTop: '5px',
}

const topLabel = css`
  top: calc(${form.labelOffset} - (${form.paddingTop} + 10px));
  font-size: 12px;
`

export const FormGroupStyled = styled.div`
  width: 100%;
  font-size: 15px;
  position: relative;
  display: inline-flex;
  flex-direction: column; 
  color: ${({ theme }) => theme.form.label};
  padding: ${form.labelOffset} 0 10px 0;

  label {
    position: absolute;
    display: block;
    top: calc(${form.labelOffset} + ${form.paddingTop});
    left: 0px;
    transition: 0.2s ease all;
    pointer-events: none;
    cursor: text;
  }

  input {
    :focus ~ label,
    :not([value=""]) ~ label {
      ${topLabel};
    }
  }

  select,
  div {
    :focus ~ label,
    :not(.empty) ~ label {
      ${topLabel};
    }
  }

  input,
  select {
    :focus ~ label {
      color: ${({ theme }) => theme.focus};
    }
  }

  ${({ theme, invalid }) => invalid && css`
    input,
    select,
    input:focus,
    select:focus {
      border-color: ${theme.colors.danger};
    }

    input:focus ~ label,
    select:focus ~ label {
      color: inherit;
    }
  `};
`

export const ErrorStyled = styled.span`
  color: ${({ theme }) => theme.colors.danger};
  font-size: 13px;
`
