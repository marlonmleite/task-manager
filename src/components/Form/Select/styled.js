import styled from 'styled-components'
import { InputComponent } from '../Input'

const Select = InputComponent.withComponent('select')

const SelectStyled = styled(Select)`
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
`

export default SelectStyled
