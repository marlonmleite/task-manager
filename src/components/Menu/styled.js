import styled, { css } from 'styled-components'
import { Link as LinkRouter } from 'react-router-dom'
import Home from 'core/assets/svg/home.svg'
import Task from 'core/assets/svg/task.svg'
import Tag from 'core/assets/svg/tag.svg'

const icon = css`
  width: 24px;
  height: 24px;
  padding-right: 24px;
`

export const NavMenu = styled.nav`
  width: 232px;
  min-width: 232px;
  height: 100%;
`

export const Nav = styled.ul`
  list-style: none;
  margin: 24px 8px 12px 8px;
  padding: 0;
`

export const NavItem = styled.li`
  display: flex;
  align-items: center;
  cursor: pointer;
  color: #616162;

  :hover {
    background-color: rgba(0,0,0,.03);
    border-radius: 2px;
    color: #212121;
  }

  ${({ active }) => active && css`
    background-color: rgba(0,0,0,.05);
    border-radius: 2px;
    color: #212121;
    font-weight: bold;

    &:hover {
      background-color: rgba(0,0,0,.05);
    }
  `}
}
`

export const Link = styled(LinkRouter)`
  display: flex;
  align-items: center;
  flex: 1;
  color: inherit;
  font-size: 14px;
  line-height: 40px;
  text-decoration: none;
  padding: 0 16px;
`

export const HomeIcon = styled(Home)`
  fill: #e2ac4a;
  ${icon};
`

export const TaskIcon = styled(Task)`
  fill: #4285f4;
  ${icon};
`

export const TagIcon = styled(Tag)`
  fill: #55bf20;
  ${icon};
`
