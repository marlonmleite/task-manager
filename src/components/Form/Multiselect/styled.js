import Select from 'react-select'
import styled from 'styled-components'

const SelectStyled = styled(Select)`
  .multiselect {
    &__control {
      border-color: #e5e5e5;
      border-style: solid;
      border-width: 0 0 1px 0;
      border-radius: 0;
      background-color: transparent;
      min-height: 38px;

      &--is-focused {
        box-shadow: none;
        border-color: ${({ theme }) => theme.focus};
      }
    }

    &__value-container {
      padding: 5px 5px 5px 0;
    }

    &__menu {
      border-radius: 2px;
    }
  }
`

export default SelectStyled
