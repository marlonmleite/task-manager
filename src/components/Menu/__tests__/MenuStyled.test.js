import React from 'react'
import { shallow } from 'enzyme'
import { Nav, NavMenu, NavItem, Link, HomeIcon, TaskIcon, TagIcon } from '../styled'

describe('MenuStyled', () => {
  it('should render NavMenu', () => {
    const nav = shallow(<NavMenu />)

    expect(nav).toMatchSnapshot()
  })

  it('should render Nav', () => {
    const nav = shallow(<Nav />)

    expect(nav).toMatchSnapshot()
  })

  it('should render NavItem', () => {
    const nav = shallow(<NavItem active />)

    expect(nav).toMatchSnapshot()
  })

  it('should render Link', () => {
    const link = shallow(<Link active />)

    expect(link).toMatchSnapshot()
  })

  it('should render HomeIcon', () => {
    const icon = shallow(<HomeIcon />)

    expect(icon).toMatchSnapshot()
  })

  it('should render TaskIcon', () => {
    const icon = shallow(<TaskIcon />)

    expect(icon).toMatchSnapshot()
  })

  it('should render TagIcon', () => {
    const icon = shallow(<TagIcon />)

    expect(icon).toMatchSnapshot()
  })
})
