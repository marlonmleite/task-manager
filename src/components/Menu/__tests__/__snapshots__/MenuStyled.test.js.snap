// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuStyled should render HomeIcon 1`] = `
.c0 {
  fill: #e2ac4a;
  width: 24px;
  height: 24px;
  padding-right: 24px;
}

<jest-stub
  className="c0"
/>
`;

exports[`MenuStyled should render Link 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  color: inherit;
  font-size: 14px;
  line-height: 40px;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0 16px;
}

<Link
  active={true}
  className="c0"
  replace={false}
/>
`;

exports[`MenuStyled should render Nav 1`] = `
.c0 {
  list-style: none;
  margin: 0;
  padding: 0;
}

<ul
  className="c0"
/>
`;

exports[`MenuStyled should render NavItem 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  color: #616162;
  background-color: rgba(0,0,0,.05);
  border-radius: 2px;
  color: #212121;
  font-weight: bold;
}

.c0:hover {
  background-color: rgba(0,0,0,.03);
  border-radius: 2px;
  color: #212121;
}

.c0:hover {
  background-color: rgba(0,0,0,.05);
}

<li
  className="c0"
/>
`;

exports[`MenuStyled should render NavMenu 1`] = `
.c0 {
  width: 232px;
  min-width: 232px;
  height: 100%;
}

<nav
  className="c0"
/>
`;

exports[`MenuStyled should render TagIcon 1`] = `
.c0 {
  fill: #55bf20;
  width: 24px;
  height: 24px;
  padding-right: 24px;
}

<jest-stub
  className="c0"
/>
`;

exports[`MenuStyled should render TaskIcon 1`] = `
.c0 {
  fill: #4285f4;
  width: 24px;
  height: 24px;
  padding-right: 24px;
}

<jest-stub
  className="c0"
/>
`;
